### PR TITLE
Expanded search space

### PR DESCRIPTION
**Pull request by martyn**
Expands the random search options.

### CODE REVIEW

File: hypergan/search/random_search.py [hunk 1]
```diff
@@ -5,116 +5,82 @@
...
-        discriminators = overrides['discriminators']
-        model = overrides['model']
-        batch_size = overrides['batch_size']
         self.options = {
-            'batch_size':batch_size,
-            'model':model,
             'trainer': self.trainers(),
-            'discriminators': discriminators,
-            'generator': generators,
+            #'discriminators': discriminators,
```
Comment (mikkel) [inline hypergan/search/random_search.py]: remove comment

File: hypergan/search/random_search.py [hunk 2]
```diff
@@ -5,116 +5,82 @@
...
-            'g_momentum': [0,0.1,0.01,1e-6,1e-5,1e-1,0.9,0.999, 0.5],
-            'd_momentum': [0,0.1,0.01,1e-6,1e-5,1e-1,0.9,0.999, 0.5],
-            'd_decay': [0.8, 0.9, 0.99,0.999,0.995,0.9999,1],
-            'g_decay': [0.8, 0.9, 0.99,0.999,0.995,0.9999,1],
-            'd_rho': [0.99,0.9,0.95,0.1,0.01,0],
-            'g_rho': [0.99,0.9,0.95,0.1,0.01,0],
-            'd_initial_accumulator_value': [0.99,0.9,0.95,0.1,0.01],
-            'g_initial_accumulator_value': [0.99,0.9,0.95,0.1,0.01],
+        selector = hc.Selector({
+            'd_learn_rate': self.range(),
```
Comment (mikkel) [inline hypergan/search/random_search.py]: I wonder if we want the range to be smaller on learn rate.  Something like self.range / 100.0

File: hypergan/search/random_search.py [hunk 3]
```diff
@@ -5,116 +5,82 @@
...
+            'd_initial_gradient_squared_accumulator_value': self.range(),
+            'g_initial_gradient_squared_accumulator_value': self.range(),
             'd_clipped_weights': False,
             'clipped_gradients': False,
             'd_trainer':tftrainers,
-            'g_trainer':tftrainers
-        }
+            'g_trainer':tftrainers,
+            'create': [
+                hg.trainers.proportional_control_trainer.create,
```
Comment (mikkel) [inline hypergan/search/random_search.py]: do mixing these trainer create / runs make sense?

Comment (martyn) [inline hypergan/search/random_search.py]: nope!  This will be fixed after #84

**Pull request merged**